### DISCUSSION TITLE
Only activate a new language if most of the strings are translated.

### DISCUSSION THREAD

**Discussion by almet** [spiral-project/ihatemoney#1196] [General]
Ihatemoney is translated in many languages, and it's really great to see this.

31 languages are listed in the `default_settings.py`, and 36 languages [are present in Weblate](https://hosted.weblate.org/projects/i-hate-money/i-hate-money/).

For a bunch of these, the translation started, but not all the strings are translated. For instance, we currently have 13 languages for which less than 25% of the strings are translated.

I'm not sure what we should do with this. Should we include only languages when they reach a certain percentage of the strings translated? Only when this reaches 100%?

To me, the whole translation process takes place kinda magically: some people do the translation when new strings appears — which is great — but I don't know how to reach them or how to make them participate in these discussions.

Do you have any idea about this?

Cheers!

**Comment by almet** [comment]
My take on this would be to raise a bit the bar of the default integrated languages. After all, it's in the default settings, and it can be tweaked by admins if they want to.

Including only the languages with all strings translated would make sense in terms of stability: It probably should be easy to get the  missing strings for all the languages which have more than 85% of the strings translated (13 languages at the time of writing), and for the rest it seems more complex.

But I guess it would make an incentive for languages which are not integrated yet, in order to reach 100% of the strings. I'm curious what other people think about it?

**Comment by zorun** [comment]
I usually look at Weblate before a release and add languages that have a decent coverage.

For 6.0.0, I went a bit further than usual, I added all languages with at least 15%. The rationale is that if a language is available on the main instance ihatemoney.org, and people notice some missing translations, they will be motivated to help translating. We can discuss this and then document the consensus for future releases.

To communicate with translators: Weblate has a feature to post announcements and optionally send an email to all translators, I did this recently for the 6.0.0 release.

**Reply by almet** [reply]
Okay, thanks for the update on this. Making announcements on Weblate seems a good way to let them know, indeed.

Thanks for explaining the rationale. It makes sense, and I believe we have to chose between quality of the end product and motivating people to translate. To me, 15% isn't enough, and I would advocate for a larger ratio, because I want to have a good quality software in the end, and half-translated language seems a non-goal. I would prefer to have less languages enabled but with a better coverage.

Of course, I'm open to discussion on this. Just want to clear things up :-)
